spyderlib/widgets/sourcecode/codeeditor.py

```diff
@@ -519,13 +530,72 @@ def create_shortcuts(self):
         unblockcomment = create_shortcut(self.unblockcomment, context='Editor',
                                          name='Unblockcomment', parent=self)
 
+        def cb_maker(attr):
+            move_type = getattr(QTextCursor, attr)
+            return functools.partial(self.cursor_move_event, move_type)
+
+        line_start = create_shortcut(cb_maker('StartOfLine'), context='Editor',
+                                     name='Start of line', parent=self)
+        line_end = create_shortcut(cb_maker('EndOfLine'), context='Editor',
+                                   name='End of line', parent=self)
+
+        prev_line = create_shortcut(cb_maker('Up'), context='Editor',
+                                    name='Previous line', parent=self)
+        next_line = create_shortcut(cb_maker('Down'), context='Editor',
+                                    name='Next line', parent=self)
+
+        prev_char = create_shortcut(cb_maker('Left'), context='Editor',
+                                    name='Previous char', parent=self)
+        next_char = create_shortcut(cb_maker('Right'), context='Editor',
+                                    name='Next char', parent=self)
+
+        prev_word = create_shortcut(cb_maker('StartOfWord'), context='Editor',
+                                    name='Previous word', parent=self)
+        next_word = create_shortcut(cb_maker('EndOfWord'), context='Editor',
+                                    name='Next word', parent=self)
+
+        kill_line_end = create_shortcut(self.kill_line_end, context='Editor',
+                                        name='Kill to line end', parent=self)
+        kill_line_start = create_shortcut(self.kill_line_start,
+                                          context='Editor',
+                                          name='Kill to line start',
+                                          parent=self)
+
+        yank = create_shortcut(self._kill_ring.yank, context='Editor',
+                               name='Yank', parent=self)
+        kill_ring_rotate = create_shortcut(self._kill_ring.rotate,
+                                           context='Editor',
+                                           name='Rotate kill ring',
+                                           parent=self)
+
+        kill_prev_word = create_shortcut(self.kill_prev_word, context='Editor',
+                                         name='Kill previous word',
+                                         parent=self)
+        kill_next_word = create_shortcut(self.kill_next_word, context='Editor',
+                                         name='Kill next word', parent=self)
+
+        start_doc = create_shortcut(cb_maker('Start'), context='Editor',
+                                    name='Start of Document', parent=self)
+
+        end_doc = create_shortcut(cb_maker('End'), context='Editor',
+                                  name='End of document', parent=self)

```

Comment:
I think going to the beginning and end of the doc should be fixed, but I'm not sure.